sklearn/svm/_base.py

```diff
@@ -115,6 +141,15 @@ def __init__(
         self.max_iter = max_iter
         self.random_state = random_state
 
+    @classmethod
+    def _filter_validation_params(cls, params_to_filter):
+        # used by subclasses if they don't use all params from here
+        return {
+            param: constraint
+            for param, constraint in cls._parameter_constraints.items()
+            if param not in params_to_filter
+        }
```

Comment:
Until now we build the inherited dict like this
```
_parameter_constraints = {
    **BaseClass._parameter_constraints,
    <new parameters>
}
_parameter_constraints.pop(<param>)
```

Let's keep doing that for consistency. This is not a bad idea though and we might consider it but if we do we'd rather implement it in BaseEstimator such that every estimator can use it (but let's leave that for a later PR).

Reply:
Will do &ndash; happy to work on that PR when the time comes :smile:

Reply:
Removing this seems to annoy `mypy`, which I haven't used before. How do you suggest I fix the issue?

```
sklearn/svm/_classes.py:1043: error: Cannot determine type of "_parameter_constraints"
sklearn/svm/_classes.py:1259: error: Cannot determine type of "_parameter_constraints"
sklearn/svm/_classes.py:1468: error: Cannot determine type of "_parameter_constraints"
sklearn/svm/_classes.py:1667: error: Cannot determine type of "_parameter_constraints"
```

Reply:
Yes it's a known issue with mypy which should be fixed in newer versions of mypy. For now we ignore this error. I updated my snippet accordingly